### DISCUSSION TITLE
how to get a dh-pages branch

### DISCUSSION THREAD

**Discussion by WraithLyd** [squidfunk/mkdocs-material#5662] [Q&A] [answered]
I have followed the `https://squidfunk.github.io/mkdocs-material/` and hosted the codes in github, but I didn't get a `gh-pages` branch. So I can't publish my site via `gh-pages` branch

**Comment by squidfunk** [comment, marked as answer]
In general, if you just follow our [publishing your site guide](https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-mkdocs), the branch should be set up for you. If that doesn't work, it might be a good idea to consul [GitHub's documentation](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site).
